pkg/operator/controller/controller.go

```diff
@@ -934,3 +929,45 @@ func (r *ReconcileVMImportConfig) isMutable(obj runtime.Object) bool {
 	}
 	return false
 }
+
+func (r *ReconcileVMImportConfig) customizeControllerConfiguration(namespace string) error {
+	configMap := corev1.ConfigMap{}
+	configMapID := client.ObjectKey{Namespace: namespace, Name: ctrlConfig.ConfigMapName}
+	if err := r.client.Get(context.TODO(), configMapID, &configMap); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		if namespacedName := r.getOsMappingConfigMapName(namespace); namespacedName != nil {
+			configMap.Data = make(map[string]string)
+			configMap.Data[ctrlConfig.OsConfigMapNameKey] = namespacedName.Name
+			configMap.Data[ctrlConfig.OsConfigMapNamespaceKey] = namespacedName.Namespace
+		}
+		configMap.Name = configMapID.Name
+		configMap.Namespace = configMapID.Namespace
+		if err = r.client.Create(context.TODO(), &configMap); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileVMImportConfig) getOsMappingConfigMapName(namespace string) *types.NamespacedName {
+	var configMapName, configMapNamespace string
+	operatorDeployment := &appsv1.Deployment{}
+	key := client.ObjectKey{Namespace: namespace, Name: "vm-import-operator"}
+	if err := r.client.Get(context.TODO(), key, operatorDeployment); err == nil {
+		operatorEnv := operatorDeployment.Spec.Template.Spec.Containers[0].Env
+		for _, env := range operatorEnv {
```

Comment:
Can you add an event  / warn that the deployment contains env-vars which are no longer supported and it was replaced by the config-map ?
We probably can't edit or remove the env variables from an existing deployment, but we should somehow let the user be aware that the configuration contains an obsolete vars (assuming the vm-import-operator is installed by the HCO CSV).

Reply:
Fixed.